extensions/objects/templates/FilepathEditor.js

```diff
@@ -73,27 +162,321 @@ oppia.directive('filepathEditor', [
+        };
+
+        $scope.getCroppingAreaDynamicStyles = function() {
+          var styles = {};
+
+          // Position and size.
+          styles.left = $scope.croppingArea.x1 + 'px';
+          styles.top = $scope.croppingArea.y1 + 'px';
+
+          var cropWidth = $scope.croppingArea.x2 - $scope.croppingArea.x1;
+          var cropHeight = $scope.croppingArea.y2 - $scope.croppingArea.y1;
+          styles.width = cropWidth + 'px';
+          styles.height = cropHeight + 'px';
+
+          // Cursor.
+          var position = $scope.mousePositionWithinCroppingArea;
+          styles.cursor = CROP_CURSORS[position];
+          if (!styles.cursor) {
+            styles.cursor = 'default';
+          }
+
+          // Translucent background layer.
+          if ($scope.isUserCropping()) {
+            var data = 'url(' + $scope.data.metadata.uploadedImageData + ')';
+            styles.background = data + ' no-repeat';
+
+            var x = $scope.croppingArea.x1 + 3;  // Add crop area border.
+            var y = $scope.croppingArea.y1 + 3;  // Add crop area border.
+            styles['background-position'] = '-' + x + 'px -' + y + 'px';
+
+            var dimensions = $scope.calculateTargetImageDimensions();
+            styles['background-size'] = dimensions.width + 'px ' +
+                                        dimensions.height + 'px';
+          }
+
+          return Object.keys(styles).map(
+            function(key) {
+              return key + ': ' + styles[key];
+            }).join('; ');
+        };
+
+        $scope.getUploadedImageDynamicStyles = function() {
+          var dimensions = $scope.calculateTargetImageDimensions();
+          var w = dimensions.width;
+          var h = dimensions.height;
+          return 'width: ' + w + 'px; height: ' + h + 'px;';
+        };
+
+        $scope.confirmCropImage = function() {
+          // Find coordinates of the cropped area within original image scale.
+          var dimensions = $scope.calculateTargetImageDimensions();
+          var r = $scope.data.metadata.originalWidth / dimensions.width;
+          var x1 = $scope.croppingArea.x1 * r;
+          var y1 = $scope.croppingArea.y1 * r;
+          var width = ($scope.croppingArea.x2 - $scope.croppingArea.x1) * r;
+          var height = ($scope.croppingArea.y2 - $scope.croppingArea.y1) * r;
+
+          // Generate new image data and file.
+          var newImageData = getCroppedImageData(
+              $scope.data.metadata.uploadedImageData,
+              x1, y1, width, height);
+
+          var newImageFile = convertImageDataToImageFile(newImageData);
+
+          // Update image data.
+          $scope.data.metadata.uploadedFile = newImageFile;
+          $scope.data.metadata.uploadedImageData = newImageData;
+          $scope.data.metadata.originalWidth = width;
+          $scope.data.metadata.originalHeight = height;
+
+          // Re-calculate dimensions reset cropping area to the boundaries.
```

Comment:
Is there a missing "and" in this sentence?

Also, slight rewording suggestion: "... and reset the boundaries of the cropping area."